sdk-design-doc-template.md

```diff
@@ -0,0 +1,83 @@
+    - `Type1 var1` : description of input variable.
+        - Make note of any additional features to know about this input, such as min/max values, etc.
+    - `Type2 var2` : description of input variable.
+
+### New API #2
+
+…
+
+---
+
+## Updated APIs
+
+APIs that already exist in the SDKs that are getting updated in some way (generally fields are being added). A brief explanation of why the API is being updated as well should be provided.
+
+### Updated API #1
+
+- `Type newProperty` : description of the new property.
+    - Make note of any additional features to know about the new property, such as min/max values, required/optional, etc.
+    - Provide generic examples of setting/getting the new property in code.
+        - `NewApi setProperty(Type prop)`
+        - `Type getProperty()`
+- `NewApi function(Type1 var1, Type2 var2,...)` : description of what the function does.
+    - Make note of any additional features/ramifications of using this function (anything that may not be immediately clear to the user).
+    - `Type1 var1` : description of input variable.
+        - Make note of any additional features to know about this input, such as min/max values, required/optional, etc.
+    - `Type2 var2` : description of input variable.
+
+### Updated API #2
+
+…
+
+---
+
+## Internal Changes
+
+If applicable, description of any internal changes to the SDKs that may impact usage that don't involve an API change. It should describe why the change is happening and its relevancy to the feature. It should describe any new expected behavior and how it differs from old behavior, as well as how users should remedy the behavior chagne.
+
+### Response Codes
+
+If applicable, include any new consensus node response codes associated with this feature. Response codes are messages that the consensus node returns when a transaction succeeds or fails.
+
+#### Transaction Retry
+
+The SDK, in some cases, needs to understand when to retry on a given response code. Evaluate the new features and identify if there are specific cases the SDK may need to consider retrying. 
+
+## Test Plan
+
+Create a numbered list of end-to-end tests that should be implemented to fully test the new feature design. These should use the Given/When/Then idiom in order to fully define the test prerequisites and setup, what’s being tested, and the expected response. The tests should invoke all new response codes and validate them.
+
+1. Given <prerequisites>, when <action>, then <result>.
+2. Given <prerequisite #1>, <prerequisite #2>, and <prerequisite #3>, when <action>, then <result #1>, <result #2>, and <result #3>.
+3. …
+
+### TCK
+
+When adding new SDK features, consider whether a tests should be defined in the TCK repository. The next step would be to create issues in the TCK repository based on this content to document all end-to-end tests and link it as reference in the design doc.
```

Comment:
My guess is that every test in the test plan must be implemented in the TCK. Next to that additional tests can make sense?

Reply:
Agreed, updated the language here.